include/hwinfo.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file
+ *
+ * @brief Public APIs to get device Information.
+ */
+
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_HWINFO_H_
+#define ZEPHYR_INCLUDE_HWINFO_H_
+
+/**
+ * @brief Device ID
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include <sys/types.h>
+#include <device.h>
+
+typedef ssize_t (*hwinfo_get_device_id_t)(struct device *dev,
+					 u8_t *buffer, size_t length);
+
+struct hwinfo_device_id_driver_api {
+	hwinfo_get_device_id_t get_id;
+};
```

Comment:
This should be:
```
struct hwinfo_driver_api {
	hwinfo_get_device_id_t get_device_id;
};
```

Reply:
This piece of code could have also been written as
```
struct hwinfo_driver_api {
	ssize_t (*get_device_id)(struct device *dev, u8_t *buffer, size_t length);
};
```
which is more compact and - I believe - easier to read. The typedefs are not really needed. Both styles are currently used in Zephyr API so the choice depends on a personal preference.

Reply:
@MaureenHelm Thx.
Done